bullet/src/EntityManagementFeatures.cc

```diff
@@ -216,36 +281,99 @@ Identity EntityManagementFeatures::GetEngineOfWorld(const Identity &) const
 std::size_t EntityManagementFeatures::GetModelCount(
     const Identity &) const
 {
-  return 0;
+  return this->models.size();
 }
 
 Identity EntityManagementFeatures::GetModel(
-    const Identity &, std::size_t) const
+    const Identity &, std::size_t _modelIndex) const
 {
-  return this->GenerateIdentity(0);
+  std::size_t index = 0;
+  std::size_t modelEntity = 0;
+  ModelInfoPtr modelPtr = nullptr;
+
+  for (auto itModel = this->models.begin();
+       itModel != this->models.end();
+       ++itModel)
+  {
+    if (index++ == _modelIndex)
+    {
+      modelPtr = itModel->second;
+      modelEntity = itModel->first;
+      break;
+    }
+  }
+
+  return this->GenerateIdentity(modelEntity, modelPtr);
 }
 
 Identity EntityManagementFeatures::GetModel(
-    const Identity &, const std::string &) const
+    const Identity &, const std::string &_modelName) const
 {
-  return this->GenerateIdentity(0);
+  std::size_t modelEntity = 0;
+  ModelInfoPtr modelPtr = nullptr;
+  for (auto itModel = this->models.begin();
+       itModel != this->models.end();
+       ++itModel)
+  {
+    if (itModel->second->name == _modelName)
+    {
+      modelPtr = itModel->second;
+      modelEntity = itModel->first;
+      break;
+    }
+  }
+
+  return this->GenerateIdentity(modelEntity, modelPtr);
 }
 
 const std::string &EntityManagementFeatures::GetModelName(
-    const Identity &) const
+    const Identity &_modelID) const
 {
-  static const std::string modelName = "bulletModel";
+  // Check if the model exists
+  if (this->models.find(_modelID.id) == this->models.end())
+  {
+    static const std::string modelName = "";
+    return modelName;
+  }
+
+  static const std::string modelName = this->models.at(_modelID)->name;
```

Comment:
I know it was this way before, but why the use of `static` here?

Reply:
I think this must be a copy/paste error from line 335. It kind of makes sense on line 335 but seems definitely wrong to use here since it means every single valid model will always report having the same name, and that name will be whichever was the first one to be asked for it.

Reply:
I think this should work: https://github.com/gazebosim/gz-physics/pull/385

Basically, if it's in the map, return the reference, otherwise return a static `std::string` that will have a stable reference.  We should be able to follow this pattern in the other places where we do this.

Returning a string reference is not a great API for this reason, but I suppose it's a bit late for that.